dev/wlp-gradle/java.gradle

```diff
@@ -31,9 +31,9 @@ subprojects {
       }
       // For Java 9+ builds, set the --release compiler flag to keep bootclasspath in sync/ 
       if(JavaVersion.current().isJava9Compatible() && !'off'.equals(bnd('javac.args.release'))) {
-        def release = sourceCompatibility
-        if(sourceCompatibility.startsWith('1.'))
-          release = sourceCompatibility.substring(2);
+        def release = targetCompatibility
```

Comment:
since most builds use Java 11 now, this change will have an effect. The bundles that you have changed to specify source=1.7 target=1.8 will be compiled with --release=8 which will allow them to use new Java 8 language features and be compiled with Java 8 class libraries, which is what Alasdair wanted to avoid here.

Reply:
The only way to get around this is to make sure things build correctly in eclipse which has source 7 checking, build with Java 8 instead of Java 11, or build Java 7 and then build Java 8 after that so that you make sure that it works for both.  I think it is a risk we can take and we can monitor changes using eclipse and clearly explain to developers to not use Java 8 language features until we are ready to say we are not going back to Java 7 for any reason.